app/modules/sightings/models.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+"""
+Sightings database models
+--------------------
+"""
+
+from app.extensions import FeatherModel, db
+import uuid
+
+
+class Sighting(db.Model, FeatherModel):
+    """
+    Sightings database model.
+    """
+
+    guid = db.Column(
+        db.GUID, default=uuid.uuid4, primary_key=True
+    )  # pylint: disable=invalid-name
+    title = db.Column(db.String(length=50), nullable=False)
+
+    def __repr__(self):
+        return (
+            '<{class_name}('
+            'guid={self.id}, '
+            "title='{self.title}'"
+            ')>'.format(class_name=self.__class__.__name__, self=self)
+        )
+
+    @db.validates('title')
+    def validate_title(self, key, title):  # pylint: disable=unused-argument,no-self-use
+        if len(title) < 3:
+            raise ValueError('Title has to be at least 3 characters long.')
+        return title
+
+    def get_owners(self):
+        owners = []
+        for encounter in self.get_encounters():
+            if encounter.get_owner() is not None and encounter.get_owner() not in owners:
+                owners.append(encounter.get_owner())
+        return owners
+
+    def get_owner(self):
+        #this is what we talked about but it makes me squeamish
```

Comment:
Agreed, I would certainly suggest a len != 0 check first

Reply:
I've now added that check, good idea. I meant more that I'm still a little uncomfortable with the simplicity... If we can have any number of owners, I'm not sure if it will ever be safe to just pick one off the top of the stack. Maybe additional logic to return the one with ownership of the most encounters on the sighting? Willing to discuss.

Reply:
Ah yes, I see what you mean. 
I think your idea of returning the owner of the most encounters is a sensible one. Sightings are different in that they can potentially have multiple owners.
Actually could they? Does this cause security issues in terms of 
User A creates encounter, with 3 sightings.
User B adds 4 of their sightings to the encounter and hence becomes the owner.
User B deletes encounter as it is now the owner.
Do we want to prevent a different user from the owner adding a sighting. This needs a bit more thought. What do you allow in old world.

In terms of the implementation, get_owners is only called from this method so I'd remove it and move the encounters list iteration into here and figure out the owner here

Reply:
using "owner" in the sense of multi-owners (by way of sub-objects: encoutners) is a new concept for nextgen.  as such, i am not sure how to proceed -- but have some ideas.

i can see that we probably(!) pretty clearly want the .owner property to be the (original) creator of the sighting.  but right now the only place i can imagine caring about owner-by-way-of-encounters has to do with **security**, no?  so maybe multi-ownership (in terms of encounter-depth ownership) should be left to be resolved in _security-related_ methods on sightings?

in other words, we can imagine, say, making a decision about "who can delete a sighting?"  would this be based perhaps just one logic around .owner?  or would we want to somehow traverse the encounter-owners?  maybe sighting _read_ privilege would be based on our typical algorithms (based on .owner) but grant additional access if you happen to be an encounter-owner?  does that make sense?

in this way, the `get_owners()` method might really only be of interest "internally" for security checks.

Reply:
though now i see the `user.owns_object(sighting)` is a little tricky based on what i say in previous comment.  for that to be an outright boolean concept (outside of context of "owner in what way?") is a tough call.  i am totally fine with it proceeding as written: basically mimicking old-world idea that if you owned an encounter _in a sighting_, thats good enough.  (i just honestly dont remember if that was "good enough" _to do anything_ to said sighting -- or if it was used only for _reading_ ?   anyway, lets just keep it as that for now and deal with later if we need something more subtle.

Reply:
In "old wildbook" we don't let you delete a sighting at all... it just hangs out until every encounter it contains is either removed or deleted then it just goes away. I think there is a great argument for that since it respects every participating user.